src/core/address.c

```diff
@@ -70,53 +70,57 @@ uint8_t address_serialized_len(address_t *addr) {
   return 1 + address_len(addr);
 }
 
-int address_serialize(address_t *addr, byte_t bytes[], size_t len) {
+size_t address_serialize(address_t *addr, byte_t bytes[], size_t len) {
   // validate parameters
   if (addr == NULL || bytes == NULL) {
-    return -1;
+    return 0;
   }
 
-  bytes[0] = (uint8_t)addr->type;
   // check buffer len
   if (len < address_serialized_len(addr)) {
-    return -1;
+    return 0;
   }
 
+  bytes[0] = (uint8_t)addr->type;
   memcpy(bytes + 1, addr->address, address_len(addr));
-  return 0;
+  return sizeof(uint8_t) + address_len(addr);
 }
 
-int address_deserialize(byte_t bytes[], size_t len, address_t *addr) {
+size_t address_deserialize(byte_t bytes[], size_t len, address_t *addr) {
```

Comment:
Should `address_deserialize` also create a new address with `malloc` like other deserialization functions?